src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets

```diff
@@ -206,6 +209,28 @@
       OutputListLocation="$(OutDir)assemblylist.txt"
       Condition="'$(UseLegacyXunitPerfRunner)'=='true'"
      />
+     
+     <!-- For Desktop Tests, we need to generate Binding Redirects. -->
+     <ItemGroup>
+        <_FileNamesInReferenceCopyLocalPaths Include="@(ReferenceCopyLocalPaths->'%(Filename)%(Extension)')">
+          <SourcePath>%(ReferenceCopyLocalPaths.Identity)</SourcePath>
+        </_FileNamesInReferenceCopyLocalPaths>
+        <_FileNamesInTestCopyLocalByFileNameWithoutDuplicates Include="@(_TestCopyLocalByFileNameWithoutDuplicates->'%(Filename)%(Extension)')">
+          <SourcePath>%(_TestCopyLocalByFileNameWithoutDuplicates.SourcePath)</SourcePath>
+        </_FileNamesInTestCopyLocalByFileNameWithoutDuplicates>
+        <_FilteredListForBindingRedirect Include="@(_FileNamesInReferenceCopyLocalPaths)" Exclude="@(_FileNamesInTestCopyLocalByFileNameWithoutDuplicates)" />
+        <_FilteredListForBindingRedirect Include="@(_FileNamesInReferenceCopyLocalPaths)" />
```

Comment:
I'm not sure what this accomplishes
```
+        <_FilteredListForBindingRedirect Include="@(_FileNamesInReferenceCopyLocalPaths)" Exclude="@(_FileNamesInTestCopyLocalByFileNameWithoutDuplicates)" />
 +        <_FilteredListForBindingRedirect Include="@(_FileNamesInReferenceCopyLocalPaths)" />
```

It includes one list excepting some things, then includes it again entirely.